### PR TITLE
CSPL-1458: Fixing failing C3 appframework test

### PR DESCRIPTION
**Pull request by jambrosiano**
- Scaling test: Adding missing app upload for C3
- Scaling test: Removing unnecessary v1 apps upload
- Scaling test: removing redundant lines
- Redistributing smoke and integration tests
- Changing nodes instance type to m5.2xlarge

Test runs:

Upgrade:
[2] • [SLOW TEST:1728.911 seconds]
[2] c3appfw test
[2] /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:29
[2]   Single Site Indexer Cluster with SHC (C3) with App Framework
[2]   /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:55
[2]     smoke, c3, appframework: can deploy a C3 SVA with App Framework enabled, install apps and upgrade them
[2]     /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:56
[2] ------------------------------
[2] {"level":"info","ts":1636749462.917367,"msg":"testenv deleted.\n","testenv":"c3appfw-fc1"}
[2] {"level":"info","ts":1636749462.917418,"msg":"testenv deleted.\n","testenv":"c3appfw-fc1"}
[2]
[2] JUnit report was created: /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/c3appfw-fc1_junit.xml
[2]
[2] Ran 1 of 1 Specs in 1743.493 seconds
[2] SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
[2] PASS

Downgrade:
[1] • [SLOW TEST:1827.498 seconds]
[1] c3appfw test
[1] /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:29
[1]   Single Site Indexer Cluster with SHC (C3) with App Framework
[1]   /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:267
[1]     smoke, c3, appframework: can deploy a C3 SVA with App Framework enabled, install apps and downgrade them
[1]     /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:268
[1] ------------------------------
[1] {"level":"info","ts":1636760231.0194829,"msg":"testenv deleted.\n","testenv":"c3appfw-y03"}
[1] {"level":"info","ts":1636760231.019532,"msg":"testenv deleted.\n","testenv":"c3appfw-y03"}
[1]
[1] JUnit report was created: /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/c3appfw-y03_junit.xml
[1]
[1] Ran 1 of 2 Specs in 1842.695 seconds
[1] SUCCESS! -- 1 Passed | 0 Failed | 1 Pending | 0 Skipped
[1] PASS

Scale:
[1] • [SLOW TEST:3449.966 seconds]
[1] c3appfw test
[1] /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:29
[1]   Single Site Indexer Cluster with SHC (C3) with App Framework
[1]   /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:475
[1]     integration, c3, appframework: can deploy a C3 SVA with App Framework enabled, install apps,scale up IDXC and SHC, install app on new pods, scale down
[1]     /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/appframework_test.go:476
[1] ------------------------------
[1] {"level":"info","ts":1636770996.111288,"msg":"testenv deleted.\n","testenv":"c3appfw-kp2"}
[1] {"level":"info","ts":1636770996.1113129,"msg":"testenv deleted.\n","testenv":"c3appfw-kp2"}
[1]
[1] JUnit report was created: /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/c3/appframework/c3appfw-kp2_junit.xml
[1]
[1] Ran 1 of 3 Specs in 3465.817 seconds
[1] SUCCESS! -- 1 Passed | 0 Failed | 2 Pending | 0 Skipped
[1] PASS

### CODE REVIEW

**Review by tpereirasplunk — APPROVED** [review]

File: test/c3/appframework/appframework_test.go [hunk 1]
```diff
@@ -192,9 +190,16 @@ var _ = Describe("c3appfw test", func() {
 			managerPodNames := []string{fmt.Sprintf(testenv.ClusterManagerPod, deployment.GetName()), fmt.Sprintf(testenv.DeployerPod, deployment.GetName())}
```
Comment (pdhanoyasplunk) [inline test/c3/appframework/appframework_test.go:190]: `managerPodNames` is same as `podNames`

File: test/c3/appframework/appframework_test.go [hunk 2]
```diff
@@ -1339,7 +1309,7 @@ var _ = Describe("c3appfw test", func() {
 	})
 
 	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
-		It("smoke, c3, appframework: can deploy a C3 SVA instance with App Framework enabled and install above 200MB of apps at once", func() {
+		It("integration, c3, appframework: can deploy a C3 SVA instance with App Framework enabled and install above 200MB of apps at once", func() {
```
Comment (pdhanoyasplunk) [inline test/c3/appframework/appframework_test.go:1312]: Cleanup lIne 1388 by declaring shReplicas := 3
and replace code on 1405 with allpodNames

**Review by pdhanoyasplunk — CHANGES_REQUESTED** [review]

File: test/c3/appframework/appframework_test.go [hunk 3]
```diff
@@ -357,44 +371,49 @@ var _ = Describe("c3appfw test", func() {
...
-			allPodNames := testenv.DumpGetPods(testenvInstance.GetName())
+			allPodNames := podNames
+			allPodNames = append(allPodNames, testenv.GeneratePodNameSlice(testenv.SearchHeadPod, deployment.GetName(), shReplicas, false, 1)...)
+			allPodNames = append(allPodNames, testenv.GeneratePodNameSlice(testenv.IndexerPod, deployment.GetName(), indexerReplicas, false, 1)...)
+
 			testenvInstance.Log.Info("Verify apps are copied to correct location based on Pod KIND for app", "version", appVersion)
 			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), allPodNames, appListV2, true, true)
 
+			testenvInstance.Log.Info("Verify apps are copied to correct location for MC", "version", appVersion)
+			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), []string{mcPodName}, appListV2, true, true)
```
Comment (pdhanoyasplunk) [inline test/c3/appframework/appframework_test.go:403]: clusterwide set to true for MC verification should be false.

File: test/c3/appframework/appframework_test.go [hunk 4]
```diff
@@ -357,44 +371,49 @@ var _ = Describe("c3appfw test", func() {
...
 
-			// Verify apps are installed on MC and C3(cluster-wide)
-			testenvInstance.Log.Info("Verify apps are installed on the pods by running Splunk CLI commands for app", "version", appVersion)
+			// Verify apps are installed on C3
+			testenvInstance.Log.Info("Verify V2 apps are installed on C3 Indexers and SHs", "version", appVersion)
 			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), allPodNames, appListV2, true, "enabled", true, true)
 
+			// Verify apps are installed on MC
+			testenvInstance.Log.Info("Verify V2 apps are installed on MC", "version", appVersion)
+			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), []string{mcPodName}, appListV2, true, "enabled", true, true)
```
Comment (pdhanoyasplunk) [inline test/c3/appframework/appframework_test.go:415]: Clusterwide set to true for Monitoring console

File: test/c3/appframework/appframework_test.go [hunk 5]
```diff
@@ -439,86 +459,59 @@ var _ = Describe("c3appfw test", func() {
...
 
-			// Verify apps are downgraded on C3 (cluster-wide)
-			testenvInstance.Log.Info("Verify apps are downgraded on the pods by running Splunk CLI commands for app", " version", appVersion)
+			// Verify apps are downgraded on C3(cluster-wide)
+			testenvInstance.Log.Info("Verify apps are downgraded on the C3 pods", "version", appVersion)
 			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), allPodNames, appListV1, true, "enabled", false, true)
+
+			// Verify apps are downgraded on MC
+			testenvInstance.Log.Info("Verify apps are downgraded on MC", "version", appVersion)
+			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), []string{mcPodName}, appListV1, true, "enabled", false, true)
```
Comment (pdhanoyasplunk) [inline test/c3/appframework/appframework_test.go:478]: Clusterwide set to true for MC.

File: test/c3/appframework/appframework_test.go (hunk 2 above)
Comment (pdhanoyasplunk) [reply test/c3/appframework/appframework_test.go:1312]: I see the code on line 1405 is still there .. We have function to generate these 
```
allPodNames = append(allPodNames, testenv.GeneratePodNameSlice(testenv.SearchHeadPod, deployment.GetName(), shReplicas, false, 1)...)
			
allPodNames = append(allPodNames, testenv.GeneratePodNameSlice(testenv.IndexerPod, deployment.GetName(), indexerReplicas, false, 1)...)
```

File: test/c3/appframework/appframework_test.go [hunk 6]
```diff
@@ -1433,18 +1397,11 @@ var _ = Describe("c3appfw test", func() {
...
-			for i := 0; i < int(shReplicas); i++ {
-				sh := fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), i)
-				podNames = append(podNames, string(sh))
-			}
+			podNames = append(podNames, testenv.GeneratePodNameSlice(testenv.SearchHeadPod, deployment.GetName(), shReplicas, false, 1)...)
+			podNames = append(podNames, testenv.GeneratePodNameSlice(testenv.IndexerPod, deployment.GetName(), indexerReplicas, false, 1)...)
 
 			// Verify apps are installed on indexers and SH
-			testenvInstance.Log.Info("Verify apps are installed on the pods by running Splunk CLI commands")
+			testenvInstance.Log.Info("Verify apps are installed on the Indexers and Searc Heads")
```
Comment (pdhanoyasplunk) [inline test/c3/appframework/appframework_test.go:1404]: NIT: `Searc Heads` vs `Search Heads`

**Review by pdhanoyasplunk — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]

**Pull request merged**